pkg/trace/api/otlp.go

```diff
@@ -638,15 +638,15 @@ func (o *OTLPReceiver) convertSpan(rattr map[string]string, lib pcommon.Instrume
 func resourceFromTags(meta map[string]string) string {
 	// `http.method` was renamed to `http.request.method` in the HTTP stabilization from v1.23.
 	// See https://opentelemetry.io/docs/specs/semconv/http/migration-guide/#summary-of-changes
-	if _, m := getFirstFromMap(meta, "http.request.method", "http.method"); m != "" {
+	if _, m := getFirstFromMap(meta, semconv125.AttributeHTTPRequestMethod, semconv125.AttributeHTTPMethod); m != "" {
 		// use the HTTP method + route (if available)
 		if _, route := getFirstFromMap(meta, semconv.AttributeHTTPRoute, "grpc.path"); route != "" {
 			return m + " " + route
 		}
 		return m
 	} else if m := meta[string(semconv.AttributeMessagingOperation)]; m != "" {
 		// use the messaging operation
-		if _, dest := getFirstFromMap(meta, semconv.AttributeMessagingDestination, semconv117.AttributeMessagingDestinationName); dest != "" {
+		if _, dest := getFirstFromMap(meta, semconv.AttributeMessagingDestination, semconv125.AttributeMessagingDestinationName); dest != "" {
```

Comment:
The [messaging doc](https://github.com/open-telemetry/semantic-conventions/blob/v1.25.0/docs/messaging/messaging-spans.md) says:

> Warning Existing messaging instrumentations that are using [v1.24.0 of this document](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/messaging/messaging-spans.md) (or prior) SHOULD NOT change the version of the messaging conventions that they emit until a transition plan to the (future) stable semantic conventions has been published. Conventions include, but are not limited to, attributes, metric and span names, and unit of measure.

One solution for this is to move to v1.24.0 instead of v1.25.0, since that one does not have yet this warning. We can also check if these two conventions have changed and leave an explicit comment on this part of the code saying that we shouldn't update these until things become stable in this namespace.

Reply:
Actually there is no semconv 1.24. The available version before 1.25 is 1.22, while 1.22 does not have the HTTP method or status attributes.

I'm OK with closing this PR and wait until messaging conventions become stable. WDYT?

Reply:
We can also contribute v1.24.0 upstream, up to you!

Reply:
Wasn't able to add 1.24 to collector following the instructions
```
% make gensemconv SPECPATH=/Users/yang.song/go/src/github.com/open-telemetry//semantic-conventions SPECTAG=v1.24.0
...
opentelemetry.semconv.model.exceptions.ValidationError: Semantic Convention attributes.http.common reference `http.request.method` but it cannot be found! - @2:5 ('attributes.http.common')
```

Will close this for now.